### PR TITLE
txmempool: add thread safety annotations

### PR DESCRIPTION
**Pull request by ajtowns**
Add missing thread safety guards to CTxMempool members.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

File: src/txmempool.h [hunk 1]
```diff
@@ -479,21 +479,21 @@ class CTxMemPool
 protected:
     const int m_check_ratio; //!< Value n means that 1 times in n we check.
     std::atomic<unsigned int> nTransactionsUpdated{0}; //!< Used by getblocktemplate to trigger CreateNewBlock() invocation
-    CBlockPolicyEstimator* minerPolicyEstimator;
+    CBlockPolicyEstimator* minerPolicyEstimator GUARDED_BY(cs);
```
Comment (vasild) [inline src/txmempool.h:482]: This is only ever modified by assigning it in the constructor, can be `const` like `m_check_ratio`.
```suggestion
    CBlockPolicyEstimator* const minerPolicyEstimator;
```

File: src/test/fuzz/tx_pool.cpp [hunk 2]
```diff
@@ -21,8 +21,9 @@ std::vector<COutPoint> g_outpoints_coinbase_init_mature;
 std::vector<COutPoint> g_outpoints_coinbase_init_immature;
 
 struct MockedTxPool : public CTxMemPool {
-    void RollingFeeUpdate()
+    void RollingFeeUpdate() EXCLUSIVE_LOCKS_REQUIRED(!cs)
     {
+        LOCK(cs);
```
Comment (hebasto) [inline src/test/fuzz/tx_pool.cpp:26]: Why this lock is required?

**Review by hebasto — APPROVED** [review]
ACK 05d319dc9ed2a33f2a08a698dc6d8b74ff77229b, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by practicalswift** [comment]
Concept ACK

Thanks for making the thread-safety annotations more complete!

File: src/txmempool.h (hunk 1 above)
Comment (ajtowns) [reply src/txmempool.h]: And the object protects itself via its own mutex. Done.

**Review by hebasto — APPROVED** [review]
re-ACK 793b2682841b0bdd7eb93163e34728765cfe52b2, only suggested changes since my [previous](https://github.com/bitcoin/bitcoin/pull/22003#pullrequestreview-664529633) review.

**Comment by MarcoFalke** [comment]
cr ACK 793b2682841b0bdd7eb93163e34728765cfe52b2

**Review by vasild — COMMENTED** [review]
ACK 793b2682841b0bdd7eb93163e34728765cfe52b2

**Pull request merged**